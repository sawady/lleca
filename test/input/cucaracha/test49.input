
fun main() {
    i := 0
    s := 0
    j := True
    while i < 1000 {
        if j {
            s := s + i
        }
        j := not j
        i := i + 1
    }
    putNum(s)
    putChar(10)
}
