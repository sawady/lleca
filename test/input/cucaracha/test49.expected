Cons(
  Function(
    main,
    Nil,
    Unit,
    Cons(
      StmtAssign(
        i,
        ExprConstNum(
          0
        )
      ),
      Cons(
        StmtAssign(
          s,
          ExprConstNum(
            0
          )
        ),
        Cons(
          StmtAssign(
            j,
            ExprConstBool(
              True
            )
          ),
          Cons(
            StmtWhile(
              ExprLt(
                ExprVar(
                  i
                ),
                ExprConstNum(
                  1000
                )
              ),
              Cons(
                StmtIf(
                  ExprVar(
                    j
                  ),
                  Cons(
                    StmtAssign(
                      s,
                      ExprAdd(
                        ExprVar(
                          s
                        ),
                        ExprVar(
                          i
                        )
                      )
                    ),
                    Nil
                  ),
                  Nothing
                ),
                Cons(
                  StmtAssign(
                    j,
                    ExprNot(
                      ExprVar(
                        j
                      )
                    )
                  ),
                  Cons(
                    StmtAssign(
                      i,
                      ExprAdd(
                        ExprVar(
                          i
                        ),
                        ExprConstNum(
                          1
                        )
                      )
                    ),
                    Nil
                  )
                )
              )
            ),
            Cons(
              StmtCall(
                putNum,
                Cons(
                  ExprVar(
                    s
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      10
                    ),
                    Nil
                  )
                ),
                Nil
              )
            )
          )
        )
      )
    )
  ),
  Nil
)
